PPI channel assignment for Bluetooth controller is incorrect for nRF52805

**Describe the bug**
e603b9d59e84f45285e4e0afb26d3930bed70629 has changed, among others, the following line:
```diff
@@ -350,7 +350,7 @@ static inline void hal_sw_switch_timer_clear_ppi_config(void)
  * 2 adjacent PPIs (8 & 9) and 2 adjacent PPI groups are used for this wiring;
  * <index> must be 0 or 1. <offset> must be a valid TIMER CC register offset.
  */
-#define HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_BASE 8
+#define HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_BASE 9
 #define HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI(index) \
 	(HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_BASE + index)
 #define HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_REGISTER_EVT(chan) \
```

Consequently, `hal_radio_group_task_disable_ppi_setup()` that is called from `radio_tmr_start()` and contains:
https://github.com/zephyrproject-rtos/zephyr/blob/8e69dafccf25bc5b4015a607cdba114216c4ea0e/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_ppi.h#L619-L624
will try to configure the PPI channel 10, which does not exist in nRF52805.

**To Reproduce**
Build e.g. the `peripheral_hr` sample for `nrf52dk_nrf52805` and see the following warning:
```
[160/213] Building C object zephyr/subsys/blueto...ller.dir/ll_sw/nordic/hal/nrf5/radio/radio.c.obj
In file included from ../../../../subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_ppi.h:10,
                 from ../../../../subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5.h:54,
                 from ../../../../subsys/bluetooth/controller/ll_sw/nordic/hal/radio_vendor_hal.h:10,
                 from ../../../../subsys/bluetooth/controller/./hal/radio.h:7,
                 from /home/andy/zephyrproject/zephyr/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c:19:
/home/andy/zephyrproject/zephyr/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c: In function 'radio_tmr_start':
/home/andy/zephyrproject/modules/hal/nordic/nrfx/hal/nrf_ppi.h:446:14: warning: array subscript 10 is above array bounds of 'volatile PPI_CH_Type[10]' [-Warray-bounds]
  446 |     p_reg->CH[(uint32_t) channel].EEP = eep;
      |     ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
In file included from /home/andy/zephyrproject/modules/hal/nordic/nrfx/mdk/nrf.h:129,
                 from /home/andy/zephyrproject/modules/hal/nordic/nrfx/drivers/nrfx_common.h:41,
                 from /home/andy/zephyrproject/modules/hal/nordic/nrfx/nrfx.h:38,
                 from /home/andy/zephyrproject/modules/hal/nordic/nrfx/hal/nrf_rtc.h:37,
                 from /home/andy/zephyrproject/zephyr/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c:11:
/home/andy/zephyrproject/modules/hal/nordic/nrfx/mdk/nrf52805.h:1706:21: note: while referencing 'CH'
 1706 |   __IOM PPI_CH_Type CH[10];                     /*!< (@ 0x00000510) PPI Channel                                                */
      |                     ^~
In file included from ../../../../subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_ppi.h:10,
                 from ../../../../subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5.h:54,
                 from ../../../../subsys/bluetooth/controller/ll_sw/nordic/hal/radio_vendor_hal.h:10,
                 from ../../../../subsys/bluetooth/controller/./hal/radio.h:7,
                 from /home/andy/zephyrproject/zephyr/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c:19:
/home/andy/zephyrproject/modules/hal/nordic/nrfx/hal/nrf_ppi.h:447:14: warning: array subscript 10 is above array bounds of 'volatile PPI_CH_Type[10]' [-Warray-bounds]
  447 |     p_reg->CH[(uint32_t) channel].TEP = tep;
      |     ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
In file included from /home/andy/zephyrproject/modules/hal/nordic/nrfx/mdk/nrf.h:129,
                 from /home/andy/zephyrproject/modules/hal/nordic/nrfx/drivers/nrfx_common.h:41,
                 from /home/andy/zephyrproject/modules/hal/nordic/nrfx/nrfx.h:38,
                 from /home/andy/zephyrproject/modules/hal/nordic/nrfx/hal/nrf_rtc.h:37,
                 from /home/andy/zephyrproject/zephyr/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c:11:
/home/andy/zephyrproject/modules/hal/nordic/nrfx/mdk/nrf52805.h:1706:21: note: while referencing 'CH'
 1706 |   __IOM PPI_CH_Type CH[10];                     /*!< (@ 0x00000510) PPI Channel                                                */
      |                     ^~
```

**Impact**
I don't have the actual nRF52805 to confirm this but I suspect that the Bluetooth controller won't work properly on this chip (on the nRF52 DK where nRF52805 is emulated on nRF52832, which has more programmable PPI channels, this cannot be checked).

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.12.3
 - Commit 8e69dafccf25bc5b4015a607cdba114216c4ea0e

--- a/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_ppi.h
+++ b/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_ppi.h
@@ -379,10 +379,22 @@ static inline void hal_sw_switch_timer_clear_ppi_config(void)
 
 /* Wire a SW SWITCH TIMER EVENTS_COMPARE[<cc_offset>] event
  * to a PPI GROUP TASK DISABLE task (PPI group with index <index>).
- * 2 adjacent PPIs (8 & 9) and 2 adjacent PPI groups are used for this wiring;
+ * 2 adjacent PPIs (9 & 10) and 2 adjacent PPI groups are used for this wiring;
  * <index> must be 0 or 1. <offset> must be a valid TIMER CC register offset.
  */
+#if defined(CONFIG_SOC_NRF52805)
+/* Because nRF52805 has limited number of programmable PPI channels,
+ * tIFS Trx SW switching on this SoC can be used only when pre-programmed
+ * PPI channels are also in use, i.e. when TIMER0 is the event timer.
+ */
+#if (EVENT_TIMER_ID == 0)
+#define HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_BASE 2
+#else
+#error "tIFS Trx SW switch can be used on this SoC only with TIMER0 as the event timer"
+#endif
+#else /* -> !defined(CONFIG_SOC_NRF52805) */
 #define HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_BASE 9
+#endif
 #define HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI(index) \
 	(HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_BASE + (index))
 #define HAL_SW_SWITCH_GROUP_TASK_DISABLE_PPI_REGISTER_EVT(chan) \
@@ -398,12 +410,8 @@ static inline void hal_sw_switch_timer_clear_ppi_config(void)
  * 2 adjacent PPI groups are used for this wiring. 'index' must be 0 or 1.
  */
 #if defined(CONFIG_SOC_NRF52805)
-/* Because nRF52805 has limited number of programmable PPI channels,
- * tIFS Trx SW switching on this SoC can be used only when pre-programmed
- * PPI channels are also in use, i.e. when TIMER0 is the event timer.
- */
 #if (EVENT_TIMER_ID == 0)
-#define HAL_SW_SWITCH_GROUP_TASK_ENABLE_PPI 2
+#define HAL_SW_SWITCH_GROUP_TASK_ENABLE_PPI 9
 #else
 #error "tIFS Trx SW switch can be used on this SoC only with TIMER0 as the event timer"
 #endif
@@ -418,12 +426,12 @@ static inline void hal_sw_switch_timer_clear_ppi_config(void)
 /*Enable Radio at specific time-stamp:
  * wire the SW SWITCH TIMER EVENTS_COMPARE[<cc_offset>] event
  * to RADIO TASKS_TXEN/RXEN task.
- * 2 adjacent PPIs (11 & 12) are used for this wiring; <index> must be 0 or 1.
+ * 2 adjacent PPIs (12 & 13) are used for this wiring; <index> must be 0 or 1.
  * <offset> must be a valid TIMER CC register offset.
  */
 #if defined(CONFIG_SOC_NRF52805)
 #if (EVENT_TIMER_ID == 0)
-#define HAL_SW_SWITCH_RADIO_ENABLE_PPI_BASE 3
+#define HAL_SW_SWITCH_RADIO_ENABLE_PPI_BASE 4
 #else
 #error "tIFS Trx SW switch can be used on this SoC only with TIMER0 as the event timer"
 #endif

